genet/cli.py

```diff
@@ -0,0 +1,1504 @@
+    ensure_dir(supporting_outputs)
+
+    network = _read_network(path_to_network, projection)
+
+    logging.info("Creating elevation dictionary for network nodes")
+    elevation = network.get_node_elevation_dictionary(
+        elevation_tif_file_path=path_to_elevation.as_posix(), null_value=null_value
+    )
+
+    if save_jsons:
+        _to_json(elevation, output_dir / "node_elevation_dictionary.json")
+
+    logging.info("Validating the node elevation data")
+    report = genet.elevation.validation_report_for_node_elevation(elevation)
+    logging.info(report["summary"])
+
+    if save_jsons:
+        _to_json(report, output_dir / "validation_report_for_elevation.json")
+
+    if write_elevation_to_network:
+        logging.info("Adding node elevation as attribute to the network")
+        node_attrib_dict = {}
+        for node_id in elevation.keys():
+            elevation_value = elevation[node_id]["z"]
+            node_attrib_dict[node_id] = {"z": elevation_value}
+        network.apply_attributes_to_nodes(node_attrib_dict)
+
+        gdf_nodes = network.to_geodataframe()["nodes"]
+        gdf_nodes = gdf_nodes[["id", "z", "geometry"]]
+        save_geodataframe(
+            gdf_nodes.to_crs(EPSG4326), "node_elevation", supporting_outputs
+        )
+
+    logging.info("Creating slope dictionary for network links")
+    slope_dictionary = network.get_link_slope_dictionary(elevation_dict=elevation)
+
+    if save_jsons:
+        _to_json(slope_dictionary, output_dir / "link_slope_dictionary.json")
+
+    if write_slope_to_network:
+        logging.info("Adding link slope as an additional attribute to the network")
+        attrib_dict = {}
+        for link_id in slope_dictionary.keys():
+            slope_value = slope_dictionary[link_id]["slope"]
+            attrib_dict[link_id] = {
+                "attributes": {
+                    "slope": {
+                        "name": "slope",
+                        "class": "java.lang.String",
+                        "text": slope_value,
+                    }
+                }
+            }
+        network.apply_attributes_to_links(attrib_dict)
+
+        gdf = network.to_geodataframe()["links"]
+        df = pd.DataFrame(list(slope_dictionary.items()), columns=["id", "slope_tuple"])
+        df["slope"] = [x["slope"] for x in df["slope_tuple"]]
+        df = df[["id", "slope"]]
+        gdf_links = pd.merge(gdf, df, on="id")
+        save_geodataframe(gdf_links.to_crs(EPSG4326), "link_slope", supporting_outputs)
+
+    if write_slope_to_object_attribute_file:
+        genet.elevation.write_slope_xml(slope_dictionary, output_dir)
+
+    logging.info("Writing the updated network")
+    network.write_to_matsim(output_dir)
+
+
+@cli.command()
+@xml_file("network")
+@xml_file("schedule", False)
+@xml_file("vehicles", False)
+@projection
+@vehicle_scalings
+@output_dir
+def auto_schedule_fixes(
+    path_to_network: Path,
+    path_to_schedule: Optional[Path],
```

Comment:
this should be optional really, I don't know what I was thinking in the original script - without a schedule there is nothing to fix.. could you update this to be non optional?